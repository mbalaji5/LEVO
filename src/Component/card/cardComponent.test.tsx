import React from "react";
import { render, screen } from "@testing-library/react";
import { Card } from ".";

const props = {
  article: {
    publishedAt: "2022-01-15T17:02:41.000Z",
    summary: "summary",
    title: "",
    id: 0,
    url: "",
    imageUrl: "",
    newsSite: "",
    updatedAt: "",
    featured: false,
    launches: [],
    events: []
  }
};
describe("Should render Card component", () => {
  it("Should validate snapshot", () => {
    const { container, baseElement } = render(<Card {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("Should validate element", () => {
    render(<Card {...props} />);
    const buttonElement = screen.getByText(/Read More/i);
    expect(buttonElement).toBeInTheDocument();
    const summaryElement = screen.getByText(/summary/i);
    expect(summaryElement).toBeInTheDocument();
  });
});
