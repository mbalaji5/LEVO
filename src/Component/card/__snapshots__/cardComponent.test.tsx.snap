// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Card component Should validate snapshot 1`] = `
<body>
  <div>
    <section
      class="cardItem"
      data-testid="cardItem"
    >
      <span>
        6/1/2022
      </span>
      <section>
        <h4 />
        <p
          class="summary"
        >
          summary
        </p>
      </section>
      <button
        class="readMoreButton"
      >
        Read More
      </button>
    </section>
  </div>
</body>
`;
